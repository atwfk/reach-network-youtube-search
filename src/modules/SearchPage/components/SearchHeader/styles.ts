import styled from "styled-components";

export const StyledSearchHeader = styled.section`
  display: flex;
  justify-content: space-between;
  padding: 10px 0;
  border-bottom: 1px solid #e5e5e5;
  margin-bottom: 20px;
`;
