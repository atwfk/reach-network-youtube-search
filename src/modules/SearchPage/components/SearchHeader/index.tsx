import React from "react";
import type { FC, ReactElement } from "react";
import TotalResults from "../TotalResults";
import Filter from "../Filter";
import { StyledSearchHeader } from "./styles";

const SearchHeader: FC<{ totalResults: number }> = ({ totalResults }): ReactElement => {
  return (
    <StyledSearchHeader>
      <TotalResults totalResults={totalResults} />
      <Filter />
    </StyledSearchHeader>
  );
};

export default SearchHeader;
