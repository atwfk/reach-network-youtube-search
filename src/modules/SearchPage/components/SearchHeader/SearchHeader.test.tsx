import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import SearchHeader from ".";

afterEach(cleanup);

describe("SearchHeader Component snapshots", () => {
  it("Should render simple SearchHeader component", () => {
    const tree = renderer.create(<SearchHeader totalResults={131909} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
