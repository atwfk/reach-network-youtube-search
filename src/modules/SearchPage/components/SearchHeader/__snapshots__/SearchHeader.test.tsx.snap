// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader Component snapshots Should render simple SearchHeader component 1`] = `
.c1 {
  font-weight: 400;
}

.c2 {
  color: #767272;
  font-size: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c2 .filter-icon {
  font-size: 30px;
  margin-right: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 10px 0;
  border-bottom: 1px solid #e5e5e5;
  margin-bottom: 20px;
}

<section
  className="c0"
>
  <h3
    className="c1"
  >
    About 
    131,909
     filtered results
  </h3>
  <button
    className="c2"
    type="button"
  >
    <svg
      className="filter-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 10.5a.5.5 0 0 1 .5-.5h3a.5.5 0 0 1 0 1h-3a.5.5 0 0 1-.5-.5zm-2-3a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm-2-3a.5.5 0 0 1 .5-.5h11a.5.5 0 0 1 0 1h-11a.5.5 0 0 1-.5-.5z"
      />
    </svg>
    FILTER
  </button>
</section>
`;
